src/js/components/replenishment/CreateReplenishment.jsx

```diff
@@ -139,6 +174,33 @@ class CreateReplenishment extends Component {
     }
   }
 
+  isAllProductsSelected() {
+    const selectedProductCount = numberOfProductsSelected(this.state.values.requirements);
+    const productCount = this.state.values.requirements.length;
+    return selectedProductCount === productCount;
+  }
+
+  selectAllCode() {
+    const isAllSelected = this.isAllProductsSelected();
```

Comment:
I do not understand the idea behind this.

Reply:
the logic behind here was:
- if at the moment all rows are **selected** then do the opposite and **deselect** all rows
- else if at the moment not all rows are selected the **select** them all